### PR TITLE
Line gap for each status output

### PR DESCRIPTION
**Pull request by vv7v7**
Dear Developers,

Added a line(gap) output for each status print(when fork option is being used(child count > 1)).
This is how it looks like:

john

Also, there's an bug of status printing when user asks for it too fast(**this is not related to this commit**), for example: 
```
...
-----------------------------------------------------------------------------------
1 0g 0:00:00:01  3/3 0g/s 100965p/s 424977c/s 424977C/s 44119..55123
----------------------------------------------------------------------------------- // Here user asked for status faster than(before) last child has shown his status
1 0g 0:00:00:02  3/3 0g/s 60070p/s 253697c/s 253697C/s asaak..512221
2 0g 0:00:00:01  3/3 0g/s 60793p/s 257470c/s 257470C/s merold..321489
2 0g 0:00:00:02  3/3 0g/s 43712p/s 190864c/s 190864C/s mernher..mernent
...
```
Does anyone know how to fix this? Is that something like defining a variable which would contain output's status and if previous output has been finished then continue?

Best regards

### CODE REVIEW

**Comment by vv7v7** [comment]
Oh, *ci/cicleci* gives "misc.c:29:10: fatal error: sys/ioctl.h: No such file or directory". Can this be ignored?

**Comment by claudioandre-br** [comment]
Thanks for looking into the issue.

`ioctl` system call is not portable. You need to `#ifdef` it properly. 

Well, first, I guess you should wait for some feedback from magnum and solar. 
I'm not sure if they will appreciate the extra verbosity.

File: src/status.c [hunk 1]
```diff
@@ -298,6 +298,9 @@ static void status_print_cracking(double percent)
 		if (n > 0)
 			p += n;
 	}
+		
+	if (options.node_min == 1)
+		fprintf(stderr, "%s\n", status_gap());
```
[the remark this answers is not in the document]
Comment (magnumripper) [reply src/status.c]: Here's how to do that:
```c
#if HAVE_MPI
	if (mpi_p > 1)
		MPI_Barrier(MPI_COMM_WORLD);
#endif
```

**Comment by magnumripper** [comment]
> Also, there's an bug of status printing when user asks for it too fast (...) Does anyone know how to fix this? Is that something like defining a variable which would contain output's status and if previous output has been finished then continue?

It could be fixed but isn't worth the complexity. When forked, communication between processes is very complex. I think my proposed barrier will fix it for MPI though, as a good "side effect".

File: src/status.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (magnumripper) [reply src/status.c]: I'm not sure we'd want this line for single-process jobs though. Perhaps wrap all of the above (including the barrier) in `if (options.node_count) {` ... `}`.
```c
	if (options.node_count) {
		if (john_main_process)
			fprintf(stderr, "%s\n", status_gap());
#if HAVE_MPI
		if (mpi_p > 1)
			MPI_Barrier(MPI_COMM_WORLD);
#endif
	}
```

**Comment by vv7v7** [comment]
> 
> 
> Thanks for contributing. Perhaps this feature should be configurable in `john.conf`.

Thank you very much for your answers. Understood the changes.

**Comment by solardiz** [comment]
Reading the above comments, it looks like @magnumripper sees value in this functionality for MPI builds, where a barrier is easy. If so, I won't object to having it as an option.

Instead of a dependency on the `ioctl`, I think we should simply print empty lines (when this option is enabled) before the groups of status lines.

**Comment by magnumripper** [comment]
@vv7v7 unfortunately Solar is right - even the very delimiting line might get printed in the wrong place. I'm closing this, but don't let that discourage you for future contributions!

**Comment by solardiz** [comment]
We have a few issues labeled "good first issue":

https://github.com/magnumripper/JohnTheRipper/labels/good%20first%20issue

Perhaps we should review the issue list and add this label to some more.

@vv7v7 Please feel free to contribute by fixing any of those, or find some other issue suitable for you. Thanks!

**Pull request closed without merging**